src/index.ts

```diff
@@ -100,16 +97,37 @@ export class StatsStore {
 
   private gitHubUser: string | null;
 
-  public constructor(appName: AppName, version: string, isDevMode: boolean, getAccessToken = () => "") {
+  /** How often daily stats should be submitted (in ms). */
+  private reportingFrequency: number;
+
+  /** If true, it'll print every metrics request on the console. */
+  private verboseMode: boolean;
+
+  public constructor(
+    appName: AppName,
+    version: string,
+    isDevMode: boolean,
+    getAccessToken = () => "",
+    options: {
+      logInDevMode?: boolean,
+      reportingFrequency?: number,
+      verboseMode?: boolean,
+    } = {},
+  ) {
     this.version = version;
     this.appUrl = baseUsageApi + appName;
     const optOutValue = localStorage.getItem(StatsOptOutKey);
 
-    this.isDevMode = isDevMode;
+    this.isDevMode = !options.logInDevMode && isDevMode;
     this.getAccessToken = getAccessToken;
     this.gitHubUser = null;
 
-    this.timer = this.getTimer(ReportingLoopIntervalInMs);
+    this.reportingFrequency = options.reportingFrequency || exports.DailyStatsReportIntervalInMs;
```

Comment:
so if no `reportingFrequency` is supplied, we default back to only checking to see if we should send metrics every 24 hours.  Currently, we are checking to see if it's time to send metrics every 4 hours, but only sending metrics once every 24 hours.  

The rationale for having separate `reportingLoopInterval` and `statsReportingInterval` was in case the attempt to send daily metrics failed for some reason (central is down, the user's internet is down, the user's machine is off, etc).  I'm open to changing that, but wanted to ask about edge cases.

If metrics can't be sent when the daily reporting interval is hit, and we are only checking to see if it's time to send stats when the reporting interval is hit, is that going to skew our metrics?  Like would the metrics that didn't get sent roll into the next payload and skew our dashboards? Let me know if this doesn't make sense, happy to clarify further.

Reply:
I've tried to keep the retry logic the same as it was before for the default scenario: this means it'll send the metrics every 24h and retry it every 4 hours.

This is done in L130 (I see that GitHub is displaying the diff in a weird way), by dividing the reporting frequency by `6` (which is the same ratio as 4h every 24h). I'm gonna add a comment on the code for the future reader.

I'm using ratios since it makes more sense now that the frequency interval is configurable (e.g if the user selects to send the metrics every hour, we may want to retry every 10min).

I thought about making this parameter also configurable, but I hardly see any scenario where a user would want to customize this value so I preferred to not add more config options.

Reply:
oh ok cool, I didn't realize the logic was the same because the way the diff is displayed.  great!  Thanks for the explanation.